Mission board

datadanne opened the pull request:
Closes RIG-59, RIG-60, RIG-61, RIG-62, RIG-64

sanderpick commented on garage/src/abis/MissionsManager.ts:
would it be cleaner to have this as part of the rigs package vs. manual abi?

datadanne replied:
Yes definitely. We actually export the generated types from typechain in the rigs package but we can't use them, I don't remember if it is because they are just incompatible with wagmi or if it has something to do with the way they are exported from the rigs package but that would definitely be much nicer

dtbuchholz replied:
✅

sanderpick commented on ethereum/contracts/MissionsManager.sol:
```diff
@@ -0,0 +1,101 @@
+    ///
+    /// @param missionId  The mission id
+    /// @param data       JSON-string with the data for the submission
+    function submitMissionContribution(
```
i remember chatting about this... so the front end does some validation / construction of `data`, but ultimately `data` could be anything cause this method is open to the world. we'll just manually look at the entries and throw out any junk, right?

datadanne replied:
Yeah exactly. Contributions need to be manually reviewed by an admin and won't show in the Garage UI until they've been approved, so you can't spam a bunch of junk and make it look ugly

dtbuchholz replied:
@datadanne i verified the contract on [mumbai](https://mumbai.polygonscan.com/address/0xb169b0bed3e4ca014dc329318935e0dcf50e14b6) to test out the "anyone can write" aspect (no issues with the logic!). lmk if you want me to push the commit; it just updates the `verify.ts` script.

datadanne replied:
> @datadanne i verified the contract on [mumbai](https://mumbai.polygonscan.com/address/0xb169b0bed3e4ca014dc329318935e0dcf50e14b6) to test out the "anyone can write" aspect (no issues with the logic!). lmk if you want me to push the commit; it just updates the `verify.ts` script.

nice! yes, please push that commit :pray:

dtbuchholz replied:
✅

sanderpick reviewed (dismissed):
Whoa, tons of work here! LGTM. @dtbuchholz can you give this a skim as well? lots of additions.

dtbuchholz commented on ethereum/test/MissionsManager.ts:
```diff
@@ -0,0 +1,238 @@
+          "description",
+          JSON.stringify(["t1"]),
+          JSON.stringify(["r1"]),
+          JSON.stringify([{ amount: 1337, currency: "FT" }]),
```
👀 [1337](https://1337skulls.xyz/)

dtbuchholz commented on garage/src/hooks/usePersistentState.ts:
```diff
@@ -0,0 +1,54 @@
+): [T, React.Dispatch<React.SetStateAction<T>>] => {
+  const isLocalStorageAvailable = (() => {
+    try {
+      localStorage.setItem("test", "test");
```
should these values be "test"? or does it not really matter what they are?

datadanne replied:
yeah it doesn't matter what the values are, it is just testing if localStorage is available and falling back to a cookie if it isn't. also ht gpt for writing this

dtbuchholz commented on garage/src/components/SignManifestoModal.tsx:
```diff
@@ -0,0 +1,124 @@
+          </Flex>
+        </ModalBody>
+        <ModalFooter>
+          <Button onClick={onAgree}>I AGREE</Button>
```
when a user agrees to the terms, i don't think that data is stored in a table, correct? it'd be a nice little FT bonus we could track in the `missions` table. if you sign the pledge, you get a little FT bump—the `max_number_of_contributions` could be capped at 3000 (1:1 with the # of Rigs).

datadanne replied:
Nope, we just store it in localStorage/cookies. I agree that it would be nice to store it in tableland but I don't think it makes sense unless we can pay the gas for the txn: feels weird to limit users from being able to see the missions & contribute unless they pay for it (most have already paid to buy/mint a rig already), and some might not have eth on arbitrum so they would have to bridge first as well

dtbuchholz replied:
Fair point. It'd be cool if we supported gasless txs for this, but that obviously adds unneeded complexity.

dtbuchholz commented on garage/src/pages/Admin/MissionAdmin.tsx:
```diff
@@ -0,0 +1,385 @@
+                  expectedChain={secondaryChain}
+                  isDisabled={["querying", "success"].includes(txnState)}
+                  onClick={() =>
+                    setContributionAcceptedState(contribution.id, true, reason)
```
should the admin contribution approval process work in this PR? i tried both accepting/rejecting with the [admin wallet](https://discord.com/channels/592843512312102924/1119061737073811596/1148514993290084374) in the [admin UI](https://rigs-garage-6t8sxo896-tableland.vercel.app/admin/missions/1), and neither worked. some screenshots:
- attempting to accept (note: both the reject/accept buttons spin...not a big deal but calling it out in case it helps troubleshoot):
<img width="942" alt="Screenshot 2023-09-11 at 3 07 41 PM" src="https://github.com/tablelandnetwork/rigs/assets/13358940/3ac254f9-f808-4768-8480-444386b88377">

- error shown after waiting for the approve tx to finalize: 
<img width="583" alt="Screenshot 2023-09-11 at 3 08 21 PM" src="https://github.com/tablelandnetwork/rigs/assets/13358940/ba57b792-d3e1-4c56-bb44-5296b7b23918">


- and upon refreshing, the second row (the one i tried to approve) is still pending: <img width="619" alt="Screenshot 2023-09-11 at 3 16 53 PM" src="https://github.com/tablelandnetwork/rigs/assets/13358940/1b06959c-6217-4696-9c2e-5ffefb5e27a6">

datadanne replied:
hm yes that should work, I'll take a look

datadanne replied:
ah, I forgot to give the admin wallet permission to update the contributions table. hm, maybe I should update the contract & have approvals go through the contract instead, that way we don't have to remember to give it both the admin role and those permissions

datadanne replied:
https://mumbai.polygonscan.com/tx/0x71078318effcf5a6946db104e6fc9be84f2628d56856a1e957e0afd7b9293202

dtbuchholz replied:
Nice!

datadanne commented:
@dtbuchholz @sanderpick can I get a re-approval on this PR? the reviews got dismissed when dtb pushed a commit. I also added three commits: 
- a tiny bug fix, I noticed today that the 'contributions pending reivew' count was off because of a missing `WHERE mission_id = ?` clause in the query ([4c992c2](https://github.com/tablelandnetwork/rigs/pull/919/commits/4c992c2a3be86ab288e4eeeabdf6a9aa3226bb7e))
- I reverted the temp commit that hardcoded the staging deployment for testing purposes ([7504bf8](https://github.com/tablelandnetwork/rigs/pull/919/commits/7504bf8a529cc73952d68c30c6d82b8ccb20b0a4))
- I made a small improvement to the contribution review form so that it doesn't crash the page if the data is incorrectly formatted, thx @dtbuchholz for doing some proper testing :raised_hands:  ([d0e0aec](https://github.com/tablelandnetwork/rigs/pull/919/commits/d0e0aec6c2c151e189b6d42a177ad9c9f8375c2a))

--

The deployment won't work because this PR needs the `mission`-related tables/contracts to be part of the rigs package. Lmk how you want to go about deploying this, should I still merge it to staging or do you want to publish a pre-release of the package first @sanderpick?

sanderpick commented:
let's merge it to staging and I can work on the deployment of the package / contract / tables

datadanne commented:
> let's merge it to staging and I can work on the deployment of the package / contract / tables

Ah actually I can't haha, it requires all checks to pass. I can add back the temp change that hardcodes the staging dev into `env.ts` and update the SHIP linear issue with a check to remember to revert that? or can you override the status checks?

![image](https://github.com/tablelandnetwork/rigs/assets/656107/eb5049a9-badf-4237-8fc1-51794a588ba7)

sanderpick commented:
ah yeah. i can just do the package release now and then you can make the changes here. one sec

sanderpick commented:
pushed 0.3.3-pre0
